File: library/src/main/java/com/pengrad/telegrambot/model/stars/partner/TransactionPartnerTelegramAds.java
package com.pengrad.telegrambot.model.stars.partner;

import java.util.Objects;

public class TransactionPartnerTelegramAds extends TransactionPartner {

    public final static String TYPE = "telegram_ads";

    public TransactionPartnerTelegramAds() {
        super(TYPE);
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TransactionPartnerTelegramAds that = (TransactionPartnerTelegramAds) o;
        return Objects.equals(type(), that.type());
    }

    @Override
    public int hashCode() {
        return Objects.hash(type());
    }

    @Override
    public String toString() {
        return "TransactionPartnerTelegramAds{" +
                "type='" + type() + "\'" +
                '}';
    }

}

File: library/src/main/java/com/pengrad/telegrambot/model/stars/partner/TransactionPartnerTelegramAds.kt
package com.pengrad.telegrambot.model.stars.partner

import com.pengrad.telegrambot.model.stars.partner.TransactionPartnerType.TELEGRAM_ADS

class TransactionPartnerTelegramAds : TransactionPartner {

    override val type: String
        get() = TELEGRAM_ADS

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as TransactionPartnerTelegramAds

        return type == other.type
    }

    override fun hashCode(): Int {
        return type.hashCode()
    }

    override fun toString(): String {
        return "TransactionPartnerTelegramAds(type='$type')"
    }
}